### PR TITLE
Update nix

### PR DESCRIPTION
**Pull request by timsears**
Update nix-support.

### CODE REVIEW

**Comment by timsears** [comment]
This PR currently points to my fork of paganini-hs. (See the top of the flake.nix file.) If the [PR for paganini-hs](https://github.com/maciej-bendkowski/paganini-hs/pull/1) get merged, then this PR can be updated to point to the proper upstream registry, before this PR gets merged.

**Comment by timsears** [comment]
The todo in the previous comment is [done](https://github.com/maciej-bendkowski/boltzmann-brain/pull/23/commits/f428310d04f531916d9a6384544476624b0958ed).

File: Data/Boltzmann/Internal/Parser.hs [hunk 1]
```diff
@@ -107,8 +107,7 @@ parseFromFile
 parseFromFile p file = runParser p file <$> readFile file
 
 -- | Prints the given parsing errors.
-printError :: (Stream t, ShowErrorComponent e) => ParseErrorBundle t e -> IO a
-
+printError :: (ShowErrorComponent e, VisualStream t, TraversableStream t) => ParseErrorBundle t e -> IO a
```
Comment (maciej-bendkowski) [inline Data/Boltzmann/Internal/Parser.hs]: Why are these type changes required?

Comment (maciej-bendkowski) [reply Data/Boltzmann/Internal/Parser.hs]: I mean not just here, but throughout the PR.

File: boltzmann-brain.cabal [hunk 2]
```diff
@@ -87,7 +87,9 @@ executable bb-tune
                      , directory
                      , filepath 
                      , megaparsec
-                     , text 
+                     , text
+                     -- for testing
+                     , buffon-machines
```
Comment (maciej-bendkowski) [inline boltzmann-brain.cabal:92]: Could you please also update the `stack.yaml`?

**Review by maciej-bendkowski — CHANGES_REQUESTED** [review]
The PR seems fine to me, although I would like you to address two concerns I mention below.

File: Data/Boltzmann/Internal/Parser.hs (hunk 1 above)
Comment (timsears) [reply Data/Boltzmann/Internal/Parser.hs]: Needed them to make it compile when I ran cabal build.

File: boltzmann-brain.cabal (hunk 2 above)
Comment (timsears) [reply boltzmann-brain.cabal:92]: I'm not a stack user, but I will give it a try.

File: Data/Boltzmann/Internal/Parser.hs (hunk 1 above)
Comment (maciej-bendkowski) [reply Data/Boltzmann/Internal/Parser.hs]: It seems that these type changes are imposed by an implicit `megaparsec` update to version `9.0.0` or higher (see https://hackage.haskell.org/package/megaparsec-9.0.1/docs/Text-Megaparsec-Stream.html#t:VisualStream). This version of `megaparsec` is not even in the latest LTS resolver of stack, (see https://www.stackage.org/lts-16.27). How have you managed to build `boltzmann-brain` while keeping the `haskell-ext-src` untouched?

**Review by maciej-bendkowski — CHANGES_REQUESTED** [review]
We cannot jump to `megaparsec-9.0.0` without changing the `stack` LTS. It's our primary building tool (especially useful for stable builds). I think that we wither stick with the old version (for now), or update the LTS but make sure that nothing breaks. The first option seems easier.

**Comment by timsears** [comment]
Which version of megaparsec are you using or can you use?

**Comment by maciej-bendkowski** [comment]
It's `megaparsec 7.0.5`. You can use `stack ls dependencies` to find out what are the current dependencies given the resolver.

**Review by timsears — COMMENTED** [review]
> We cannot jump to `megaparsec-9.0.0` without changing the `stack` LTS. It's our primary building tool (especially useful for stable builds). I think that we wither stick with the old version (for now), or update the LTS but make sure that nothing breaks. The first option seems easier.

I just tried a third option: roll back the nix snapshot to make the libs closer (if not identical) to lts-14.18. `nix-shell`/`cabal build` still works fine. It's hard for me to test stack on my box, so please lmk.

File: Data/Boltzmann/Internal/Parser.hs (hunk 1 above)
Comment (timsears) [reply Data/Boltzmann/Internal/Parser.hs]: I used nixos-20.09 as a package set and used `cabal build` in the nix shell. Then I followed the ghc errors and to get compilation working. As mentioned below I rolled back the package set to nixos-20.03 which has the megaparsec version you want. BTW, nix picks up its haskell packages from stack, although strict matching is not guaranteed.

**Comment by maciej-bendkowski** [comment]
`stack build` and `stack install` are working fine now. Please let me know if I can consider the PR done and merge.

**Comment by timsears** [comment]
Very glad to hear that. I think I can improve the flake.nix file for paganini-hs a bit and it might affect this one a tiny bit, so it might be worth waiting (until probably Monday). Or, you can go ahead I can submit another PR for those changes. Either way is fine with me.

**Comment by timsears** [comment]
Not planning on further changes at the moment. Go ahead and merge.

**Comment by maciej-bendkowski** [comment]
@timsears Thanks for your contribution! :)

**Pull request merged**